torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp

```diff
@@ -127,6 +128,60 @@ RegisterOperators reg(
            };
          },
          aliasAnalysisSpecialCase()),
+     Operator(
+         "prim::AllocateSlab() -> Storage",
+         [](const Node* node) -> Operation {
+           int64_t total_size = node->i(attr::total_size);
+           auto device_type =
+               static_cast<DeviceType>(node->i(attr::device_type));
+           return [total_size, device_type](Stack* stack) {
+             auto allocator = GetAllocator(device_type);
+             auto slab = c10::Storage(
+                 c10::Storage::use_byte_size_t(),
+                 total_size,
+                 allocator,
+                 /*resizable=*/false);
+             push(stack, std::move(slab));
+           };
+         },
+         aliasAnalysisSpecialCase()),
+     Operator(
+         prim::AllocateTensor,
+         [](const Node* node) -> Operation {
+           int64_t size = node->i(attr::size);
+           int64_t offset = node->i(attr::offset);
+           auto type = node->ty(attr::profiled_type)->expect<TensorType>();
+
+           return [offset, size, type](Stack* stack) {
+             c10::Storage slab;
+             pop(stack, slab);
+             uint8_t* start = static_cast<uint8_t*>(slab.data());
+             void* src = static_cast<void*>(start + offset);
+             at::Tensor temp_tensor = at::from_blob(
+                 src,
+                 *type->sizes().concrete_sizes(),
+                 *type->strides().concrete_sizes(),
+                 at::TensorOptions(*type->device()).dtype(*type->scalarType()));
+             temp_tensor.storage().set_nbytes(size);
+             push(stack, std::move(temp_tensor));
+           };
+         },
+         aliasAnalysisSpecialCase()),
+     Operator(
+         "prim::ReleaseSlab(Storage slab, ...) -> ()",
+         [](Stack* stack) {
+           auto num_inputs = pop(stack).toInt();
+           std::vector<at::Tensor> inputs(num_inputs - 1);
+           for (int i = 0; i < num_inputs - 1; ++i) {
+             auto temp_tensor = pop(stack).toTensor();
+             temp_tensor.storage().unsafeGetStorageImpl()->reset();
+             temp_tensor.unsafeReleaseTensorImpl()->release_resources();
+           }
+           auto slab = pop(stack).toStorage();
+           //           slab.allocator()->raw_deallocate(slab.data());
```

Comment:
commented out code

Reply:
this, like the free of the temp tensors, is up for debate (explicitness vs concision)

Reply:
The current state of the interpreter is it handles the freeing of Tensors after their final use. I dont see a compelling reason to duplicate that logic

Reply:
yea that makes sense - just wasn't sure what would be "best practices". in that case the `ReleaseSlab` node is unnecessary in its entirety